tests/utils.py

```diff
@@ -0,0 +1,19 @@
+from typing import Sequence
+
+from optuna import Trial
+from optuna.distributions import CategoricalChoiceType
+
+
+class FakeTrial(Trial):
```

Comment:
This `FakeTrial` is a brilliant idea!

Reply:
Thanks! It did take me a fair deal of digging in optuna's code to get that right tbh :stuck_out_tongue: